graph_generation/test_connection_generator.py

```diff
@@ -0,0 +1,127 @@
+"""
+Module to test graph connection generation.
+The main goal is to generate similar distributions as an input config.
+Distribution comparison is done using Kolmogorov-Smirnov test.
+
+Usage:
+    python3 graph_generation/test_connection_generator.py
+"""
+
+import unittest
+from connection_generator import ConnectionGenerator
+from scipy.stats import ks_2samp
+from collections import Counter
+
+from config_params.collection_params import CollectionParams
+from config_params.dataset_params import DatasetParams
+from config_params.dataset_to_system_params import DatasetToSystemParams
+from config_params.system_params import SystemParams
+
+import random
+random.seed(1)
+
+
+class TestNxGraph(unittest.TestCase):
+    @staticmethod
+    def get_dummy_params():
+        """Creates empty parameters for connection generator initialization."""
+        p1 = DatasetParams(0, {})
+        p2 = SystemParams(0, {}, {})
+        p3 = DatasetToSystemParams({}, {}, {}, {})
+        p4 = CollectionParams({}, {}, 0, {}, 0, {}, 0)
+        return p1, p2, p3, p4
+
+    @staticmethod
+    def generate_list_by_params(list_len, list_sum):
+        """This function generates a random list of positive integers of certain length and sum."""
+        random_array = []
+        for i in range(list_len):
+            # Each element should be at least 1, so we pick random between 1 and a half of list.
+            curr_element = random.randint(1, max(list_sum // 2, 1))
+            random_array.append(curr_element)
+            list_sum -= curr_element
+
+        # Fix the sum of the list if last element was too large.
+        random_array[random_array.index(max(random_array))] += list_sum
+        return random_array
+
+    def test_get_one_to_many_connections(self):
+        """
+        Tests if method get_one_to_many_connection generates connections with given distribution.
+        To compare config parameter distribuion, and generated distribution Kolmogorov-Smirnov statistic is used.
```

Comment:
nit: reword in "Kolmogorov-Smirnov statistic is used to compare config parameter and generated distributions"?

Reply:
Okey, can rephrase